core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -317,6 +320,31 @@ public UpdateSchema unionByNameWith(Schema newSchema) {
     return this;
   }
 
+  @Override
+  public UpdateSchema setIdentifierFields(Set<String> names) {
+    identifierFields = Sets.newHashSet();
```

Comment:
When setting an instance field, we prefix with `this.` so that it is clear that we're initializing state with a larger scope than just the method.